src/app/shopping-list/shopping-list.component.ts

```diff
@@ -8,21 +9,27 @@ import { Ingredient } from '../shared/model/ingredient.model';
   templateUrl: './shopping-list.component.html',
   styleUrls: ['./shopping-list.component.sass']
 })
-export class ShoppingListComponent implements OnInit {
+export class ShoppingListComponent implements OnInit, OnDestroy {
+
   ingredients: Ingredient[];
+  private subscription: Subscription;
```

Comment:
Please, rename with more specific name. 'subscription' may lead to mistake in case there are several items to check